Its.Log.Monitoring.UnitTests/MonitoringTestDiscoveryTests.cs

```diff
@@ -391,23 +366,113 @@ public void when_a_tag_is_specific_the_casing_on_the_tag_does_not_matter()
             var response = apiClient.GetAsync("http://blammo.com/tests/staging/widgetapi/?Brooklyn=true").Result;
 
             response.ShouldSucceed();
+            response.Content.ReadAsStringAsync().Result.Should().Contain("manhattan");            
+        }
 
-            var json = response.Content.ReadAsStringAsync().Result;
+        [Test]
+        public void properties_should_not_be_shows_as_tests()
+        {
+            var response = apiClient.GetAsync("http://blammo.com/tests/staging/widgetapi/").Result;
 
-            json.Should().Contain("manhattan");            
+            response.ShouldSucceed();
+            response.Content.ReadAsStringAsync().Result.Should().NotContain("SomeProperty");
         }
 
         [Test]
-        public void properties_should_not_be_shows_as_tests()
+        public void public_void_methods_with_optional_parameters_should_be_shown_as_tests()
         {
             var response = apiClient.GetAsync("http://blammo.com/tests/staging/widgetapi/").Result;
 
             response.ShouldSucceed();
+            response.Content.ReadAsStringAsync().Result.Should().Contain("void_test_with_optional_parameters");
+        }
 
-            var json = response.Content.ReadAsStringAsync().Result;
+        [Test]
+        public void public_type_returning_methods_with_optional_parameters_should_be_shown_as_tests()
+        {
+            var response = apiClient.GetAsync("http://blammo.com/tests/staging/widgetapi/").Result;
+
+            response.ShouldSucceed();
+            response.Content.ReadAsStringAsync().Result.Should().Contain("string_returning_test_with_optional_parameters");
+        }
+
+        [Test]
+        public void public_methods_with_non_optional_parameters_should_not_be_shown_as_tests()

```

Comment:
the source of gotchas already exists today when you put any parameter on a public method.

However this is what i had planned for v1 of monitors with inputs. allowing for non-optional parameters would expand the work needed for the change as you would need to handle calls without them + test coverage.

but it would be a great v2 feature.